Neos.Flow/Tests/Unit/Http/HeadersTest.php

```diff
@@ -471,4 +481,48 @@ public function getCacheControlDirectiveReturnsTheSpecifiedDirectiveValueIfPrese
         }
         $this->assertEquals($value, $headers->getCacheControlDirective($name));
     }
+
+    /**
+     * @test
+     */
+    public function getPreparedValuesRendersStringsAsIs()
+    {
+        $headers = new Headers();
+        $headers->set('X-String', 'Some String');
+
+        $expectedResult = [
+            'X-String: Some String',
+        ];
+        $this->assertSame($expectedResult, $headers->getPreparedValues());
+    }
+
+    /**
+     * @test
+     */
+    public function getPreparedValuesRendersDateTimeInstancesAsGmtString()
+    {
+        $headers = new Headers();
+        $headers->set('X-Date', new \DateTimeImmutable('1980-12-13'));
+
+        $expectedResult = [
+            'X-Date: Sat, 13 Dec 1980 00:00:00 GMT',
+        ];
+        $this->assertSame($expectedResult, $headers->getPreparedValues());
+    }
+
+    /**
+     * @test
+     */
+    public function getPreparedValuesRendersOneHeaderPerArrayItem()
+    {
+        $headers = new Headers();
+        $headers->set('X-Array', ['some', 'array', 123]);
```

Comment:
Well, the `123` violates the `string[]` declared in the docblock, and should be rejected as well?

Reply:
"Eagle Eye Karsten" ;)
You're right.. we need to check every array item for its type. mäh

Reply:
Could be done with an `elseif ($values !== array_filter( $values, 'is_string' ))` check after the other type checks.